ffb-client-logic/src/main/java/com/fumbbl/ffb/client/IconCache.java

```diff
@@ -269,7 +264,7 @@ public void loadIconFromUrl(String pUrl) {
 			URL iconUrl = null;
 			try {
 				iconUrl = new URL(pUrl);
-				BufferedImage icon = ImageIO.read(iconUrl);
+				BufferedImage icon = ImageIO.read(new ByteArrayInputStream(UtilHttpClient.fetchBytes(pUrl)));
```

Comment:
The created stream is never closed, and will cause a memory leak. You should be using a try block to ensure that the stream is closed. Basic example of this: https://www.infoq.com/news/2010/08/arm-blocks/

In addition, this is impacted by the comment on UtilHttpClient, so refer to that comment as well.

Reply:
I believe this is resolved by adding a close. Please tell me if I forgot anything.